src/commonTest/kotlin/it/krzeminski/snakeyaml/engine/kmp/internal/SystemTest.kt

```diff
@@ -0,0 +1,39 @@
+package it.krzeminski.snakeyaml.engine.kmp.internal
+
+import io.kotest.core.spec.style.FunSpec
+import io.kotest.matchers.shouldBe
+import io.kotest.matchers.shouldNotBe
+
+class SystemTest : FunSpec({
```

Comment:
For completeness, I think it would be good to add a test for calling `identityHashCode` on an int or a string (some types that are considered primitive on all or some platforms), and to achieve it, we should make it behave the same way on all platforms.

Reply:
Right now it would cause an exception for JS (since it expects only objects). We could:
- throw the same exception on all other platforms (might make method call a bit slower);
- add a documentation about how the "identity hashcode" behaves differently on some of the platforms.

To be honest, I am not sure what would be the best way here. Probably the first one but we would need to measure performance impact for additional checks. I will take a look at the first option if no objections from your side

Reply:
The first option was my idea. Sure, let's see what the benchmarks tell us, we have them thanks to you! 😊